### ISSUE TITLE
Restoring unstaged changes fails with git < 2.35.0

### ISSUE THREAD

**Issue by scop** [evilmartians/lefthook#584]
### :wrench: Summary

Committing something with unstaged changes ends up with:
```
Couldn't restore hidden unstaged files: exit status 129
```
...and the unstaged changes are left in a patch in .git/info.

This is with git 2.34.1, up to date Ubuntu 22.

Per verbose logs, this because of use of `--allow-empty` (added in #543).
The `--allow-empty` option was added in git 2.35.0: https://github.com/git/git/blob/master/Documentation/RelNotes/2.35.0.txt#L74-L75

### Lefthook version

1.5.2

### Steps to reproduce

Commit something in a dir with unstaged changes.

### Expected results

No problems, remporarily stashed unstaged changes applied back cleanly.

### Actual results

```
Couldn't restore hidden unstaged files: exit status 129
```

### Logs / Screenshots

```bash
LEFTHOOK_VERBOSE=true git ...
[...]
│ [lefthook] cmd: [git apply -v --whitespace=nowarn --recount --unidiff-zero --allow-empty .git/info/lefthook-unstaged.patch]
│ [lefthook] dir: [...]
│ [lefthook] err: exit status 129
│ [lefthook] out: error: unknown option `allow-empty'                                        
usage: git apply [<options>] [<patch>...]                                                  
[...]
```

**Comment by mrexox** [comment]
Wow, that's bad. Thank you for creating an issue. I would never get this error, since I currently have git 2.39.

**Comment by moitias** [comment]
Also hit this with Ubuntu 20, git 2.25.1

Seems like this was merged to 1.4.10 so for a quick workaround, downgrading to 1.4.9 works; `go install go install github.com/evilmartians/lefthook@v1.4.9`.

On a quick check at [repology](https://repology.org/project/git/versions), looks like this will be hitting also Alpine up to 3.15, CentOS up to 8, Debian up to 11, Fedora up to 34, along with the already reported Ubuntus up to 22.  

ps. :wave: @scop

**Comment by moitias** [comment]
Seems like a fix is out in 1.5.4, thanks for the quick turnaround and great piece of software @mrexox !
